framework/include/restart/DataIO.h

```diff
@@ -123,6 +126,13 @@ template<typename T>
 inline void
 dataStore(std::ostream & stream, T & v, void * /*context*/)
 {
+  #ifdef LIBMESH_HAVE_CXX11_TYPE_TRAITS
+    static_assert( std::is_polymorphic<T>::value == false,
+      "Cannot serialize a class that has virtual members!\nWrite a custom dataStore() template specialization!\n\n");
+    static_assert( std::is_trivially_copyable<T>::value || std::is_same<T, Point>::value,
```

Comment:
I had a struct containing a vector at some point. I fixed all Apps I listed
to compile with this check.

On Thu, Mar 10, 2016, 4:28 PM John W. Peterson <notifications@github.com>
wrote:

> In framework/include/restart/DataIO.h
> <https://github.com/idaholab/moose/pull/6495#discussion_r55767626>:
>
> > @@ -123,6 +126,13 @@ template<typename T>
> >  inline void
> >  dataStore(std::ostream & stream, T & v, void * /*context*/)
> >  {
> > +  #ifdef LIBMESH_HAVE_CXX11_TYPE_TRAITS
> > +    static_assert( std::is_polymorphic<T>::value == false,
> > +      "Cannot serialize a class that has virtual members!\nWrite a custom dataStore() template specialization!\n\n");
> > +    static_assert( std::is_trivially_copyable<T>::value || std::is_same<T, Point>::value,
>
> It's useful, so you could use it for discovering issues of this type, but
> I don't think we can leave it on all the time, as it will prevent legit
> apps from compiling, right? Do we have any structs-containing-Points
> currently in the apps? How did you run across this issue in the first place?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/idaholab/moose/pull/6495/files#r55767626>.
>